app/stac_api/collection_summaries.py

```diff
@@ -0,0 +1,203 @@
+import logging
+
+import numpy as np
+
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+logger = logging.getLogger(__name__)
+
+
+def float_in(flt, floats, **kwargs):
+    '''
+    This function is needed for comparing floats in order to check if a
+    given float is member of a list of floats.
+    '''
+    return np.any(np.isclose(flt, floats, **kwargs))
+
+
+def update_summaries_on_asset_delete(collection, asset):
+    '''
+    updates the collection's summaries on an asset's deletion or raises
+    errors when this fails.
+    :param collection: collection, for which the summaries probably need an update
+    :param asset: asset thats being deleted
+    For all the given parameters this function checks, if the corresponding
+    parameters of the collection need to be updated. If so, they will be either
+    updated or an error will be raised, if updating fails.
+    '''
+
+    try:
+        asset_qs = asset.__class__.objects.filter(item__collection_id=collection.pk
+                                                 ).exclude(id=asset.id)
+        if bool(asset_qs):
+
+            if not asset_qs.filter(geoadmin_variant=asset.geoadmin_variant).exists():
+                collection.summaries["geoadmin:variant"].remove(asset.geoadmin_variant)
+                collection.summaries["geoadmin:variant"].sort()
```

Comment:
out of curiosity: why is the `.sort()` necessary here?

Reply:
For example it makes unit testing simpler. With the `.sort()` you know exactly in which order the variants are to be expected. But we can also drop it, I could live without it :-)

Reply:
Ah, you are right, in the delete-case there's not really a need for `.sort()` after deleting a variant. Make only sense in case of additions

Reply:
I like the sort for unit test and also for human reading read data on the API. From my understanding they are only a few different `geoadmin:variant` ? so the sort() should not have an impact on performance, which could be an issue if we would have several thousands of variant.

Reply:
OK. I'm still not used to relying on `dict`s being ordered

Reply:
I deleted the `.sort()` in this place, as it is not needed when a variant is removed. But I left it in the other cases, if that's fine for everyone.